api/enki.py

```diff
@@ -0,0 +1,542 @@
+        if necessary.
+        """
+        if not self.token:
+            self.token = self.refresh_bearer_token()
+
+        headers = dict(extra_headers)
+        headers['Authorization'] = "Bearer " + self.token
+        headers['Library'] = self.library_id
+        if exception_on_401:
+            disallowed_response_codes = ["401"]
+        else:
+            disallowed_response_codes = None
+            response = self._make_request(
+            url=url, method=method, headers=headers,
+            data=data, params=params,
+            disallowed_response_codes=disallowed_response_codes
+        )
+        if response.status_code == 401:
+            # This must be our first 401, since our second 401 will
+            # make _make_request raise a RemoteIntegrationException.
+            #
+            # The token has expired. Get a new token and try again.
+            self.token = None
+            return self.request(
+                url=url, method=method, extra_headers=extra_headers,
+                data=data, params=params, exception_on_401=True
+            )
+        else:
+            return response
+
+    def availability(self, patron_id=None, since=None, title_ids=[], strt=0, qty=2000):
+        self.log.debug ("requesting : "+ str(qty) + " books starting at econtentRecord" +  str(strt))
+        url = str(self.base_url) + str(self.availability_endpoint)
+        args = dict()
+	args['method'] = "getAllTitles"
+	args['id'] = "secontent"
+        args['strt'] = strt
+        args['qty'] = qty
+        args['lib'] = self.lib
+	response = self.request(url, params=args)
+        return response
+
+    @classmethod
+    def create_identifier_strings(cls, identifiers):
+        identifier_strings = []
+        for i in identifiers:
+            if isinstance(i, Identifier):
+                value = i.identifier
+            else:
+                value = i
+            identifier_strings.append(value)
+
+        return identifier_strings
+
+    @classmethod
+    def parse_token(cls, token):
+        data = json.loads(token)
+        return data['access_token']
+
+    def _make_request(self, url, method, headers, data=None, params=None,
+                      **kwargs):
+        """Actually make an HTTP request."""
+        return HTTP.request_with_timeout(
+            method, url, headers=headers, data=data,
+            params=params, **kwargs
+        )
+
+    def reaper_request(self, identifier):
+        self.log.debug ("Checking availability for " + str(identifier.identifier))
+        now = datetime.datetime.utcnow()
+        url = str(self.base_url) + str(self.item_endpoint)
+        args = dict()
+        args['method'] = "getItem"
+        args['recordid'] = identifier.identifier
+        args['size'] = "small"
+        args['lib'] = self.lib
+        response = self.request(url, method='get', params=args)
+        if not(response.content.startswith("{\"result\":{\"id\":\"")):
```

Comment:
You can make this easier to read by formatting it with single quotes.

`'{"result":{"id":""`

But it would make the code more reliable to parse it as JSON and check the JSON object. Otherwise small server-side changes that don't affect the meaning of the document (such as putting the ID elsewhere in the nested object) can make it look like all the books have disappeared.